chainforge/react-server/src/backend/backend.ts

```diff
@@ -587,27 +588,41 @@ export async function queryLLM(id: string,
   if (typeof cache === 'object' && cache.cache_files !== undefined) {
     const past_cache_files: Dict = cache.cache_files;
     let past_cache_filenames: Array<string> = Object.keys(past_cache_files);
-    llms.forEach(llm_spec => {
-      let found_cache = false;
-      for (const [filename, cache_llm_spec] of Object.entries(past_cache_files)) {
-        if (matching_settings(cache_llm_spec, llm_spec)) {
-          llm_to_cache_filename[extract_llm_key(llm_spec)] = filename;
-          found_cache = true;
-          break;
-        }
-      }
-      if (!found_cache) {
-        const new_filename = gen_unique_cache_filename(id, past_cache_filenames);
-        llm_to_cache_filename[extract_llm_key(llm_spec)] = new_filename;
-        cache.cache_files[new_filename] = llm_spec;
-        past_cache_filenames.push(new_filename);
+    try {    
+      llms.forEach(llm_spec => {
+          if (QueryTracker.has(id)) {
+            console.log("HIT 1")
+            return;
+          }
+          let found_cache = false;
+          for (const [filename, cache_llm_spec] of Object.entries(past_cache_files)) {
+            if (matching_settings(cache_llm_spec, llm_spec)) {
+              llm_to_cache_filename[extract_llm_key(llm_spec)] = filename;
+              found_cache = true;
+              break;
+            }
+          }
+          if (!found_cache) {
+            const new_filename = gen_unique_cache_filename(id, past_cache_filenames);
+            llm_to_cache_filename[extract_llm_key(llm_spec)] = new_filename;
+            cache.cache_files[new_filename] = llm_spec;
+            past_cache_filenames.push(new_filename);
+          }
+        }); 
+    } catch (e) {
+      if (e.message === "CANCELLED") {
+        return;
```

Comment:
Can you remove this entire edit from lines 590-625? This check is unnecessary at this stage --loading responses from cache is almost immediate, and doesn't require querying any LLMs yet.